Added values from NF Portal Dataset schema to MC2 DataCatalog valid v…

aditya-nath-sage opened the pull request:
Based on the NF Portal Dataset schema (https://github.com/nf-osi/nf-metadata-dictionary/blob/main/registered-json-schemas/PortalDataset.json), I updated annotationProperty.csv for the DataCatalog schema for MC2 with missing terms in the NF model. Terms I didn't dd that were in the NF schema:

- dataType
- diseaseFocus
- includedInDataCatalog (think DataCatalog is analagous to this in MC2 model)
- individualCount
- manifestation
- studyID

related to https://github.com/mc2-center/data-models/issues/136

Bankso committed:
Update annotationProperty files

update Parent, validation rules, add columnType for all attributes

Bankso committed:
Update mc2.model.csv

Bankso committed:
Clean up + add first round of CDE references

Added CDE Public IDs for matches identified based on an old mapping to the CCDI model.
Removed some redundant attributes and corrected a few typos

Bankso committed:
Update mc2.model.csv

Bankso committed:
Update duo.csv

Add "Pending Annotation" as a valid value

Bankso committed:
Update valid values

Bankso committed:
Make model csv

Bankso committed:
Align Level 3 Image with Level 2

While Imaging Level 3 Image files are composed of Imaging Level 2 images, it is possible for these to be shared independently, so Imaging Level 3 Image files should have the same metadata fields available as Imaging Level 2

Bankso committed:
Update mc2.model.csv

Temporarily removed string_list from columnType (replaced with string), since schematic is not yet able to handle this value

Bankso committed:
Build JSON-LD

schematic v25.8.1

Bankso committed:
Add PIP requirements.txt

Generated by: pip freeze > requirements.txt

Bankso committed:
Update build-jsonld.yml to use requirements.txt

Bankso committed:
Update mc2.model.csv

Bankso committed:
Update mc2.model.jsonld

aditya-nath-sage committed:
Added values from NF Portal Dataset schema to MC2 DataCatalog valid values list

Bankso committed:
Add attribute info, update DataCatalog model def

Bankso committed:
Update valid values

Bankso committed:
Make model CSV

Bankso committed:
Build JSON-LD

schematic 25.8.1

Bankso committed:
Update descriptions, add DatasetView Key

Bankso committed:
Formatting

Bankso committed:
Make new model CSV with updated DataCatalog

Bankso committed:
Add DataCatalog JSON schema

Valid Synapse JSON schema (registered: https://repo-prod.prod.sagebase.org//repo/v1/schema/type/registered/MC2Center-DataCatalog-12.0.8)

Modified from original schematic output (removed minItems, added types for untyped attributes, added descriptions)

Bankso committed:
Update mc2.model.jsonld

schematic 25.8.1

Bankso committed:
Fixed incorrect column name

Bankso committed:
Prep qc model CSV

Remove cross-manifest validation rules, replace with `list like`

Bankso committed:
Build QC model JSON-LD

schematic 25.8.1

Bankso committed:
Match columnType to validation rule

Resolve schematic JSON schema generator error

Bankso committed:
Create mc2.Biospecimen-v12.0.0-schema.json

aditigopalan reviewed (commented):
Great work! I have some comments, mostly that a more comprehensive PR description would be helpful: 
1. Is this documented anywhere? For terms added; include a brief mapping note (source term, rationale) in modules/dataCatalog/annotationProperty.csv or a PR comment.
2. For “not added” NF terms (dataType, diseaseFocus, includedInDataCatalog, individualCount, manifestation, studyID), document the rationale or open follow-ups.
3.  Is this a breaking change? How does this change any templates? 
4. Idk how documentation works here, but add/adjust docs: `docs/model/dataset.md` or a new `docs/model/datacatalog.md`

aditigopalan reviewed (commented):
Do docs exist already for using the QC model?

vpchung commented on modules/education/annotationProperty.csv:
```diff
@@ -1,26 +1,26 @@
-Resource Dataset Alias,"URL or persistent identifier (DOI, Synapse ID) for any dataset(s) that is intended to be used with the item. Multiple values should be provided as a comma separated list.",,,FALSE,,,,MC2/dataset,list like
-Resource Tool Link,"URL or persistent identifier (DOI, Synapse ID) for any software or tool that is intended to be used with the item. Multiple values should be provided as a comma separated list.",,,FALSE,,,,MC2/tool,list like
-EducationalResource_id,A unique primary key that enables record updates using schematic.,,,TRUE,,,,,unique
+Attribute,Description,Valid Values,DependsOn,Required,Properties,Parent,DependsOn Component,Source,Validation Rules,Unnamed: 10
```
```suggestion
Attribute,Description,Valid Values,DependsOn,Required,Properties,Parent,DependsOn Component,Source,Validation Rules,columnType
```
Possible typo?

vpchung commented on modules/sharingPlans/annotationProperty.csv:
```diff
@@ -1,24 +1,24 @@
-DSP Dataset Description,A text description of the files contained in this dataset.,,,FALSE,,,,,str
-DSP Dataset Destination,An identifier representing the repository in which this dataset is intended to be held for long term preservation.,,,FALSE,,,,,str
-DSP Dataset Metadata,The link(s) corresponding to metadata templates that should be used to record information about this data. This field will be populated by the MC2 Center after plan submission.,,,FALSE,,,,,list like
+Attribute,Description,Valid Values,DependsOn,Required,Properties,Parent,DependsOn Component,Source,Validation Rules,Unnamed: 10
```
```suggestion
Attribute,Description,Valid Values,DependsOn,Required,Properties,Parent,DependsOn Component,Source,Validation Rules,columnType
```

vpchung reviewed (approved):
Found two instances where the `columnType` colname was not used, not sure if that would impact downstream processes, so just highlighted them for just in case!

Bankso committed:
Update modules/education/annotationProperty.csv

Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

Bankso committed:
Update modules/sharingPlans/annotationProperty.csv

Co-authored-by: Verena Chung <9377970+vpchung@users.noreply.github.com>

Bankso committed:
Update institution information

Bankso committed:
Update valid values

Bankso committed:
Add function to adjust columnType

A temporary fix to remove string_list as columnType, since this is not yet supported.

Bankso committed:
Build model CSV via make

Bankso committed:
Add code for qc model transform

Bankso committed:
Make qc model

Bankso commented:
> Do docs exist already for using the QC model?

@aditigopalan yep, QC model usage is described during step 2 here: https://sagebionetworks.jira.com/wiki/spaces/CCKPD/pages/4000776229/INTERNAL+Portal+Database+Syncing

Bankso committed:
Update Makefile

Bankso committed:
Update Makefile

Bankso committed:
Add correction for int validation rule

Clashes with number, the only numeric validation rule currently allowed by schematic

Bankso committed:
Update Makefile

Bankso committed:
Update mc2.model.csv

Bankso committed:
Delete release_workflow.sh

Bankso committed:
Update Makefile

Bankso committed:
Update mc2.model.csv

Bankso committed:
Add schematic generated schemas

Bankso committed:
Update mc2.model.jsonld

Bankso committed:
Update Makefile

Bankso committed:
Update mc2_qc.model.csv

Bankso committed:
Update mc2_qc.model.jsonld
